events/src/camera.rs

```diff
@@ -66,8 +102,128 @@ pub struct Capability {
 }
 
 /// Type representing rational number.
-#[derive(Serialize, Deserialize, Default, Debug, Clone)]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
 pub struct Fraction {
     pub numer: i32,
     pub denom: i32,
 }
+
+/// Camera status.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum Status {
+    Online,
+    Offline,
+}
+
+/// Local camera interface.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum LocalCameraInterface {
+    Usb,
+    Csi,
+}
+
+pub trait CameraBase {
```

Comment:
Cool, I was going to actually suggest having a trait. :) Why not name is just `Camera` though?

Reply:
`Camera` is used by enum already :)
Had hard time thinking another name for trait.